src/core/qgsfieldformatterregistry.h

```diff
@@ -0,0 +1,84 @@
+/***************************************************************************
+  qgsfieldformatterregistry.h - QgsFieldFormatterRegistry
+
+ ---------------------
+ begin                : 2.12.2016
+ copyright            : (C) 2016 by Matthias Kuhn
+ email                : matthias@opengis.ch
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+#ifndef QGSFIELDKITREGISTRY_H
+#define QGSFIELDKITREGISTRY_H
+
+#include <QHash>
+#include <QString>
+#include <QObject>
+
+class QgsFieldFormatter;
+
+/**
+ * \ingroup core
+ * The QgsFieldFormatterRegistry manages registered classes of QgsFieldFormatter.
+ * A reference to the QgsFieldFormatterRegistry can be obtained from
+ * QgsApplication::fieldKitRegistry().
+ *
+ * @note Added in QGIS 3.0
+ */
+class CORE_EXPORT QgsFieldFormatterRegistry : public QObject
+{
+    Q_OBJECT
+
+  public:
+
+    /**
+     * You should not normally need to create your own field kit registry.
+     *
+     * Use the one provided by `QgsApplication::fieldKitRegistry()` instead.
+     */
+    QgsFieldFormatterRegistry( QObject* parent = nullptr );
+    ~QgsFieldFormatterRegistry();
+
+    /**
+     * They will take precedence in order of adding them.
+     * The later they are added, the more weight they have.
+     *
+     * Ownership is transferred to the registry.
+     */
+    void addFieldFormatter( QgsFieldFormatter* kit );
+
+    /**
+     * Remove a field kit from the registry.
+     * The field kit will be deleted.
+     */
+    void removeFieldFormatter( QgsFieldFormatter* kit );
```

Comment:
It's a bit strange that remove takes the value, not the id. Especially when it deletes it.